### PR TITLE
Azimuthual Deviation with Sign

### LINKED ISSUE #16: Link azimuthal and elevation deviation should not be absolute values
The link's tx, rx, and elevation deviation use absolute values when computed. These values are then used in computing the scan loss for capacity calculations or gain for interference calculations. The use of absolute values is not correct.

## Expected Behavior

The deviation should be positive or negative depending on if the angle is in the clockwise or counter-clockwise direction. Similarly, it should be positive or negative if the angle is uptilt or downtilt (we have to look into the convention).

For scan loss, the deviation should be the bearing of the link minus the sector azimuth.
For interference, the deviation should be the bearing of the transmitting link that causes interference minus the bearing of the interfering path. The elevation deviation should be the elevation angle of the transmitting link that causes interference minus the elevation angle of the interfering path.

## Current Behavior

The horizontal deviation is positive regardless of whether the angle is clockwise or not. Elevation deviation is positive regardless of uptilt or downtilt.

## Detailed Description

### PR DESCRIPTION
**Pull request by dbeylkin**
## Prerequisites

- [x] I have read the [Contributing Guidelines](../blob/main/CONTRIBUTING.md).
- [x] I have read the [Code of Conduct](../blob/main/CODE_OF_CONDUCT.md).
- [x] If this is a non-trivial change, I have already opened an accompanying Issue.
- [x] If applicable, I have included documentation updates alongside my code changes.

## Description

Tx and Rx deviation should have +/- sign.

Use negative el dev in rx antenna gain and use differences between interfering path and tx/rx link el devs in interference calculation

## Test Plan

Updated unit tests

### CODE REVIEW

File: terragraph_planner/optimization/topology_interference.py
Comment (dbeylkin) [inline terragraph_planner/optimization/topology_interference.py:330]: @pohanhf please make sure you agree with this.

File: terragraph_planner/common/rf/link_budget_calculator.py
Comment (dbeylkin) [inline terragraph_planner/common/rf/link_budget_calculator.py:281]: @pohanhf do you agree with this change? The reason we have separate entries for tx and rx is for interference. For capacity, they should be the same but with opposite sign.

Comment (pohanhf) [reply terragraph_planner/common/rf/link_budget_calculator.py:281]: I think I agree with this change to have more accurate interference estimation. For capacity, I am a little bit confused now. For azimuth/horizontal scan, we take sector orientation into account. For elevation/vertical scan, do we have sector orientation? If all the devices are pointing to horizontal plane (as you mentioned below), then this is correct.

File: terragraph_planner/optimization/topology_interference.py
Comment (pohanhf) [reply terragraph_planner/optimization/topology_interference.py:330]: I think it is correct.

File: terragraph_planner/optimization/topology_interference.py
Comment (pohanhf) [inline terragraph_planner/optimization/topology_interference.py:340]: I am confused about this one. Why do we add a minus sign here?

File: terragraph_planner/common/rf/link_budget_calculator.py
Comment (dbeylkin) [reply terragraph_planner/common/rf/link_budget_calculator.py:281]: We do not have elevation angle for sectors, just horizontal. Is this something we should correct?

File: terragraph_planner/optimization/topology_interference.py
Comment (dbeylkin) [reply terragraph_planner/optimization/topology_interference.py:340]: Rx elevation deviation is minus tx elevation deviation. We don't store both of these values.

Comment (pohanhf) [reply terragraph_planner/optimization/topology_interference.py:340]: Ah I see. el_dev is only tx so no matter what links are, it is always tx elevation deviation, is it correct?

File: terragraph_planner/common/rf/link_budget_calculator.py
Comment (pohanhf) [reply terragraph_planner/common/rf/link_budget_calculator.py:281]: DN is always pointing toward horizontal plane (so no need to change). CNs are always pointing toward its DN. Not sure whether it is easy to include or not then.

File: terragraph_planner/optimization/topology_interference.py
Comment (dbeylkin) [reply terragraph_planner/optimization/topology_interference.py:340]: Exactly

File: terragraph_planner/common/rf/link_budget_calculator.py
Comment (dbeylkin) [reply terragraph_planner/common/rf/link_budget_calculator.py:281]: I'll see if I can make that change in a follow-up PR. Don't think it'll be too difficult.

I believe Wenteng at some point almost made that change but later undid it. I wonder why.

Comment (pohanhf) [reply terragraph_planner/common/rf/link_budget_calculator.py:281]: Probably let's check with our partners first. They might change their deployment method from time to time when they actually put the devices in the field. But I agree with you: no need to do it in this PR. Thanks.

**Review by pohanhf — APPROVED** [review]
Looks good to me. Thanks for the careful and detailed investigation and fix.

**Comment by dbeylkin** [comment]
@wuweicaifb can I get your review before landing this?

**Comment by dbeylkin** [comment]
Despite previous review, I guess after merging the earlier PR, this needs review approval again.

**Review by pohanhf — APPROVED** [review]

**Review by wuweicaifb — APPROVED** [review]

**Pull request merged**
